src/jdk.incubator.code/share/classes/jdk/incubator/code/type/ArrayType.java

```diff
@@ -80,18 +79,12 @@ public int dimensions() {
 
     @Override
     public ExternalizedTypeElement externalize() {
-        int dims = 0;
-        TypeElement current = this;
-        while (current instanceof ArrayType at) {
-            dims++;
-            current = at.componentType();
-        }
-        return new ExternalizedTypeElement("[".repeat(dims), List.of(current.externalize()));
+        return JavaTypeUtils.arrayType(componentType.externalize());
```

Comment:
This pattern is common to all types and refs:
* the `externalize` method returns an inflated externalized form built with the corresponding factory in `JavaTypeUtils`
* the `toString` method returns a flat string, obtained with the corresponding method in `JavaTypeUtils`